### PR TITLE
Feature hydration class

### PR DESCRIPTION
**Pull request by mrlobatoman**
### Please check for the following requirements:
- [x] I have reviewed this PR's changes at least once
- [x] I have added tests for the functionality of any new or updated features, if applicable
- [x] My code follows the style guidelines of this project

### Types of changes:
- [ ] Bugfix
- [x] Build
- [ ] Documentation
- [ ] Feature
- [ ] Refactor

### Describe your changes:
[//]: A hydrationRepository and hydrationRep test file were created. We can currently access the hydration data for a specific user ID. We have created a method that also calculates the average daily ounces drank by a user based on their all their recorded data. 

### What is the current behavior?
[//]: <> (Are there any issues / errors? Anything needing additional review?)
- Current Behavior:

### CODE REVIEW

File: src/hydrationRepository.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    constructor(hydrationData) {
+        this.hydrationData = hydrationData
+    }
+
+    getUserHydrationByID = (id) => this.hydrationData.filter(hydrationObj => hydrationObj.userID === id)
+
+    userHydrationAllTime = (id) => {
+        let userHydrationData = this.getUserHydrationByID(id)
+        let hydrationAverage = userHydrationData.reduce((acc, currObj) => {
+            acc += currObj.numOunces / userHydrationData.length
```
Comment (mrlobatoman) [inline src/hydrationRepository.js:11]: We want to do further testing and research as to why when iterating, the "/ userHydrationData.length" is being ignored until after the iterations are over.

File: test/hydrationRepo-test.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import { expect } from "chai";
+import UserHydration from "../src/hydrationRepository";
+
+
+describe("Hydration Repository", () => {
+    let hydrationStats
+    let hydrationRepo
```
Comment (mrlobatoman) [inline test/hydrationRepo-test.js:7]: These variables are subject to naming changes if it makes sense.

File: test/hydrationRepo-test.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          {
+            "userID": 33,
+            "date": "2023/03/30",
+            "numOunces": 62
+          },
+          {
+            "userID": 32,
+            "date": "2023/03/30",
+            "numOunces": 62
+          }];
```
Comment (mrlobatoman) [inline test/hydrationRepo-test.js:50]: This is the current testing data being used,  more might be needed for future methods.

File: src/hydrationRepository.js (hunk 1 above)
Comment (JacobMacFarlane) [reply src/hydrationRepository.js:11]: Look into reduce charts and how it would look like in chart format to further understanding

**Pull request merged**